optuna/_callbacks.py

```diff
@@ -63,6 +64,9 @@ class RetryFailedTrialCallback:
 
     When a trial fails, this callback can be used with the :class:`optuna.storage` class to
     recreate the trial in :obj:`TrialState.WAITING` to queue up the trial to be run again.
+    In case the retried trial fails again and is to be retried further,
+    the :func:`~optuna.storage.RetryFailedTrialCallback.retried_trial_number` of the next
+    recreated trial will point to the latest failed one.
```

Comment:
@HideakiImamura Sorry I've found this part is incorrect. Please wait a bit if you're gonna merge this PR until I fix it.

Reply:
Fixed!!!